### PR TITLE
feat(Form): multi onSubmit$ handlers

### PR DESCRIPTION
**Pull request by PatrickJS**
allow for an array of events for onSubmit$

also fixes https://github.com/QwikDev/qwik/issues/4908

### CODE REVIEW

File: packages/qwik-city/runtime/src/form-component.tsx [hunk 1]
```diff
@@ -53,13 +53,39 @@ export const Form = <O, I>(
   key: string | null
 ) => {
   if (action) {
+    const isArrayApi = Array.isArray(onSubmit$);
+    // if you pass an array you can choose where you want action.submit in it
+    if (isArrayApi) {
+      const actionInArray = Array.isArray(onSubmit$) && onSubmit$.includes(action.submit as any);
```
Comment (hamatoyogi) [inline packages/qwik-city/runtime/src/form-component.tsx:59]: I find this a bit confusing... there's already `isArrayApi` check, not sure it's needed again for `actionInArray`, or am i missing something?

Comment (PatrickJS) [reply packages/qwik-city/runtime/src/form-component.tsx:59]: oh yeah

**Review by hamatoyogi — COMMENTED** [review]
Might be worth adding a test as well to make sure it works and also one (currently failing, but could be skipped) for v2

**Comment by PatrickJS** [comment]
I made it additional api and in v2 we can make it slightly breaking api

**Comment by PatrickJS** [comment]
I added a test

**Comment by PatrickJS** [comment]
ok test added and all green

**Review by wmertens — APPROVED** [review]
LGTM

**Pull request merged**